tests/unit/executors/indexers/test_binary_indexer.py

```diff
@@ -30,3 +29,78 @@ def validate(req):
         d.ClearField('embedding')
     with f:
         f.search(docs_no_embedding, output_fn=validate, override_doc_id=False)
+
+
+def test_binarypb_update(tmpdir):
+    # required for creating workspace
+    metas = {
+        'is_trained': False,
+        'is_updated': False,
+        'batch_size': None,
+        'workspace': f'{tmpdir}',
+        'name': None,
+        'on_gpu': False,
+        'warn_unnamed': False,
+        'max_snapshot': 0,
+        'py_modules': None,
+        'pea_id': '{root.metas.pea_id}',
+        'separated_workspace': '{root.metas.separated_workspace}',
+    }
+    with BinaryPbIndexer('pbidx', metas=metas) as idxer:
```

Comment:
this test should be changed:

1. Using `test_metas` fixture will give what u expect
2. What is 'pbidx'
3. For now, I think we cannot `add` and `query` in the same `context managed` indexer. You should close the context after the addition, and then open a new indexer for querying, and then yet another for updating, and yet another for later query.  (I think this holds true specially for VectorIndexers and I think also for BinaryPbIndexer)